connect/runtime/src/test/java/org/apache/kafka/connect/runtime/SourceTaskOffsetCommitterTest.java

```diff
@@ -71,122 +74,74 @@ public void setup() {
                 Long.toString(DEFAULT_OFFSET_COMMIT_INTERVAL_MS));
         WorkerConfig config = new StandaloneConfig(workerProps);
         committer = new SourceTaskOffsetCommitter(config, executor, committers);
-        Whitebox.setInternalState(SourceTaskOffsetCommitter.class, "log", mockLog);
     }
 
     @SuppressWarnings("unchecked")
     @Test
     public void testSchedule() {
-        Capture<Runnable> taskWrapper = EasyMock.newCapture();
+        ArgumentCaptor<Runnable> taskWrapper = ArgumentCaptor.forClass(Runnable.class);
 
-        EasyMock.expect(executor.scheduleWithFixedDelay(
-                EasyMock.capture(taskWrapper), eq(DEFAULT_OFFSET_COMMIT_INTERVAL_MS),
+        when(executor.scheduleWithFixedDelay(
+                taskWrapper.capture(), eq(DEFAULT_OFFSET_COMMIT_INTERVAL_MS),
                 eq(DEFAULT_OFFSET_COMMIT_INTERVAL_MS), eq(TimeUnit.MILLISECONDS))
-        ).andReturn((ScheduledFuture) commitFuture);
-
-        PowerMock.replayAll();
+        ).thenReturn((ScheduledFuture) commitFuture);
 
         committer.schedule(taskId, task);
-        assertTrue(taskWrapper.hasCaptured());
         assertNotNull(taskWrapper.getValue());
         assertEquals(singletonMap(taskId, commitFuture), committers);
-
-        PowerMock.verifyAll();
     }
 
     @Test
     public void testClose() throws Exception {
         long timeoutMs = 1000;
 
         // Normal termination, where termination times out.
-        executor.shutdown();
-        PowerMock.expectLastCall();
-
-        EasyMock.expect(executor.awaitTermination(eq(timeoutMs), eq(TimeUnit.MILLISECONDS)))
-                .andReturn(false);
-        mockLog.error(EasyMock.anyString());
-        PowerMock.expectLastCall();
-        PowerMock.replayAll();
+        when(executor.awaitTermination(timeoutMs, TimeUnit.MILLISECONDS)).thenReturn(false);
 
         committer.close(timeoutMs);
 
-        PowerMock.verifyAll();
-        PowerMock.resetAll();
-
         // Termination interrupted
-        executor.shutdown();
-        PowerMock.expectLastCall();
-
-        EasyMock.expect(executor.awaitTermination(eq(timeoutMs), eq(TimeUnit.MILLISECONDS)))
-                .andThrow(new InterruptedException());
-        PowerMock.replayAll();
+        when(executor.awaitTermination(timeoutMs, TimeUnit.MILLISECONDS)).thenThrow(new InterruptedException());
 
         committer.close(timeoutMs);
 
-        PowerMock.verifyAll();
+        verify(executor, times(2)).shutdown();
     }
 
     @Test
     public void testRemove() throws Exception {
```

Comment:
We're losing some coverage guarantees by restructuring the tests like this. Right now we know for each sub-case (removing a task successfully, removing a cancelled one, and being interrupted while removing one) that we call `cancel` and `get` on the future. With the changes here, we do get guarantees that those methods are invoked at least once at some point in the `testRemove` method (thanks to strict stubbing), but we don't get finer-grained guarantees than that. And ensuring that those methods are invoked exactly when we expect is fairly valuable, so it'd be nice if we could keep that coverage.

I think we can break this test case out into four separate cases:
1. Removing a non-existent task
2. Removing a task successfully
3. Removing a task that gets cancelled
4. Removing a task and getting interrupted while doing so

And we can use a helper method to reduce duplication for the last three:
```java
private void testRemove() {
    // Try to remove an existing task
    when(taskFuture.cancel(false)).thenReturn(false);
    when(taskFuture.isDone()).thenReturn(false);
    when(taskId.connector()).thenReturn("MyConnector");
    when(taskId.task()).thenReturn(1);

    committers.put(taskId, taskFuture);
    committer.remove(taskId);
}
```

Which would make each test case fairly brief. For example, case 2 might look like this:

```java
@Test
public void testRemoveSuccess() throws Exception {
    when(taskFuture.get()).thenReturn(null);
    testRemove();
    assertEquals(Collections.emptyMap(), committers);
}
```

This way, after each test case completes, strict stubbing will ensure that the expected methods were invoked on the future at least once. We could go further and explicitly verify that they were invoked exactly once, but up to you if you'd like to add that coverage or not.

Reply:
I have split this into 4 tests as you suggested. I have a slight preference for only abstracting the setup of the mocks, but if you would prefer to have as little duplication as possible I will change it.